python/amici/ode_export.py

```diff
@@ -1653,12 +1837,41 @@ def _compute_equation(self, name: str) -> None:
             # force symbols
             self._derivative('xdot', 'p', name=name)
 
+        elif name == 'drootdt':
+            self._eqs[name] = self.eq('root').jacobian(time_symbol)
+
+        elif name == 'drootdt_total':
+            self._eqs[name] = self.eq('drootdt') * self._eqs['xdot'] + \
```

Comment:
Shouldn't the first `drootdt` here be `drootdx`?

Note that using `self._eqs['xdot']` means `drootdt` may depend on `w`, which was not the case in the matlab interface where `model.fun.xdot.sym_noopt` was used. To avoid further breaking with matlab generated code here, I would simply substitute in all expressions here, which is not the most efficient, but should hopefully be fine for most practical purposes.

Reply:
totally right for the drootdx, changed it. Will now do the non-optimized rhs thing.